aFWall/src/main/java/dev/ukanth/ufirewall/preferences/SecPreferenceFragment.java

```diff
@@ -258,6 +280,21 @@ public void onSharedPreferenceChanged(SharedPreferences sharedPreferences,
             //passOption = "p" + index;
 
             //currentPosition = index;
+
+            // check if device support fingerprint, if so check if one fingerprint already existed at least
+            if(FingerprintUtil.isAndroidSupport()){
+
+                checkFingerprintDeviceSupport();
+
+            }else{
+
+                if(itemList != null) {
+
+                    itemList.setValueIndex(0);
```

Comment:
I discovered that when user activate all current security option and then select none, all password and pattern and fingerprint shown at once, every dialog above the other.
This logic already there even before I added the fingerprint, I tested in donated version and I got the same behaviour.
Is this normal? :/